ddl/rollingback.go

```diff
@@ -157,11 +157,29 @@ func rollingbackAddColumns(t *meta.Meta, job *model.Job) (ver int64, err error)
 }
 
 func rollingbackDropColumn(t *meta.Meta, job *model.Job) (ver int64, err error) {
-	tblInfo, colInfo, err := checkDropColumn(t, job)
+	tblInfo, colInfo, idxInfos, err := checkDropColumn(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
 
+	if len(idxInfos) > 0 {
+		for _, indexInfo := range idxInfos {
+			switch indexInfo.State {
+			case model.StateWriteOnly, model.StateDeleteOnly, model.StateDeleteReorganization, model.StateNone:
+				// We can not rollback now, so just continue to drop index.
+				// In function isJobRollbackable will let job rollback when state is StateNone.
+				// When there is no index related to the drop column job it is OK, but when there has indices, we should
+				// make sure the job is not rollback.
+				job.State = model.JobStateRunning
+				return ver, nil
+			case model.StatePublic:
+				indexInfo.State = model.StatePublic
```

Comment:
This case means `indexInfo.State` is `model.StatePublic`. So we can remove this line.